test/add.js

```diff
@@ -38,222 +38,268 @@ function makeFileWithInlineSourceMap() {
-		t.equal(data.sourceMap.sources[0], 'helloworld.js', 'should add file to sources');
-		t.equal(data.sourceMap.sourcesContent[0], sourceContent, 'should add file content to sourcesContent');
-		t.deepEqual(data.sourceMap.names, ['helloWorld', 'console', 'log'], 'should add correct names');
-		t.equal(data.sourceMap.mappings, 'AAAA,YAAY;;AAEZ,SAASA,UAAU,CAAC,EAAE;CACrBC,OAAO,CAACC,GAAG,CAAC,cAAc,CAAC;AAC5B', 'should add correct mappings');
-		t.end();
-	});
-});
+		it('should not accept undefined as options argument', function(done) {
+			var file = makeFile();
+			[
+				{ type: 'undefined' }
+			].map(function(obj) {
+				sourcemaps.add(file, obj.val, function(err, file) {
+					expect(err instanceof Error && err.message === 'vinyl-sourcemap-add: Invalid argument: options').toExist();
+					done();
+				});
+			});
+		});
 
-test('add: should add a valid source map for CSS if wished', function (t) {
-	sourcemaps.add(makeFileCSS(), { identityMap: true }, function(err, data) {
-		t.ok(data, 'should pass something through');
-		t.ok(data instanceof File, 'should pass a vinyl file through');
-		t.ok(data.sourceMap, 'should add a source map object');
-		t.equal(String(data.sourceMap.version), '3', 'should have version 3');
-		t.equal(data.sourceMap.sources[0], 'test.css', 'should add file to sources');
-		t.equal(data.sourceMap.sourcesContent[0], sourceContentCSS, 'should add file content to sourcesContent');
-		t.deepEqual(data.sourceMap.names, [], 'should add correct names');
-		t.equal(data.sourceMap.mappings, 'CAAC;EACC;EACA', 'should add correct mappings');
-		t.end();
-	});
-});
+		it('should not accept null as options argument', function(done) {
+			var file = makeFile();
+			[
+				{ type: 'null', val: null }
+			].map(function(obj) {
+				sourcemaps.add(file, obj.val, function(err, file) {
+					expect(err instanceof Error && err.message === 'vinyl-sourcemap-add: Invalid argument: options').toExist();
+					done();
+				});
+			});
+		});
 
-test('add: should import an existing inline source map', function (t) {
-	sourcemaps.add(makeFileWithInlineSourceMap(), { loadMaps: true }, function(err, data) {
-		t.ok(data, 'should pass something through');
-		t.ok(data instanceof File, 'should pass a vinyl file through');
-		t.ok(data.sourceMap, 'should add a source map object');
-		t.equal(String(data.sourceMap.version), '3', 'should have version 3');
-		t.deepEqual(data.sourceMap.sources, ['test1.js', 'test2.js'], 'should have right sources');
-		t.deepEqual(data.sourceMap.sourcesContent, ['console.log(\'line 1.1\');\nconsole.log(\'line 1.2\');\n', 'console.log(\'line 2.1\');\nconsole.log(\'line 2.2\');'], 'should have right sourcesContent');
-		t.equal(data.sourceMap.mappings, 'AAAAA,QAAAC,IAAA,YACAD,QAAAC,IAAA,YCDAD,QAAAC,IAAA,YACAD,QAAAC,IAAA', 'should have right mappings');
-		t.end();
-	});
-});
+		it('should not accept empty string as options argument', function(done) {
+			var file = makeFile();
+			[
```

Comment:
No need to use an array and `.map` here. You can just replace `obj.val` with the value you are testing